src/generators/number/index.ts

```diff
@@ -1,16 +1,25 @@
-import {Chance} from 'chance';
+import { Chance } from 'chance';
 
 /**
- * Generates a random `number`, optionally between `lowerInclusive` and `upperExclusive`.
+ * Generates a random `number`, optionally between `lowerInclusive` and `upperExclusive` with `fixed` number of digits after the decimal.
  */
 const createNumberGenerator = (chance: Chance.Chance) => (
   lowerInclusive = -1000000,
   upperExclusive = 1000000,
-): number =>
-  chance.floating({
+  fixed = 4,
+): number => {
+  const EPSILON = 0.0000001;
+  const MAX_PRECISION = 6;
+
+  if (upperExclusive != null && upperExclusive > MAX_PRECISION) {
+    throw new TypeError(`"upperExclusive" must be less than or equal to ${MAX_PRECISION} or null`);
+  }
+
+  return chance.floating({
     min: lowerInclusive,
-    // chance.floating() returns 4 digits after the decimal at most, and its `max` is inclusive
-    max: upperExclusive - 0.0001,
+    max: upperExclusive - EPSILON,
```

Comment:
restore this comment? it's very much not obvious why we're subtracting this epsilon

Reply:
Addressed in 910d483